docs: Add warnings about using layer 7 LBs with TLS routing (#18823)
gravitational/teleport: docs/pages/includes/tls-multiplexing-warnings.mdx

Using Teleport's TLS routing mode behind a layer 7 (HTTP/HTTPS) proxy is generally not supported, due to
these proxies terminating TLS themselves and then rewriting their requests to the upstream service, stripping
the additional SNI/ALPN parts of the request in the process. In order for ALPN to work correctly, the Teleport
proxy must terminate TLS itself.

Broadly, this means that Teleport's TLS routing functionality is incompatible with:
- AWS ALBs (Application Load Balancers)

> - AWS NLBs (Network Load Balancers), when using a TLS listener and a public ACM (Amazon Certificate Manager) certificate

Reviewer: ```suggestion
- AWS NLBs (Network Load Balancers) when using a TLS listener and a public ACM (Amazon Certificate Manager) certificate
```

- Commonly used HTTP reverse proxies including nginx, Apache, Caddy, Traefik, HAProxy and many others
- Cloudflare tunnels in their default configuration

> Deploying Teleport in TLS routing mode behind an HTTP proxy will result in a Teleport web UI experience that seems

Reviewer: ```suggestion
Deploying Teleport in TLS routing mode behind an HTTP proxy will result in a Teleport Web UI experience that seems
```
To be consistent with the use of this as a product name in the docs

to work perfectly, but the use of `tsh`, `tctl` and attempting to join remote Teleport services to the cluster will fail

> with errors like `ssh: overflow reading version string` and `EOF`. A functioning Teleport web UI is not always an indication

Reviewer: ```suggestion
with errors like `ssh: overflow reading version string` and `EOF`. A functioning Teleport Web UI is not always an indication
```

of a correctly configured Teleport cluster.

> If in doubt, remove all load balancers/proxies from the equation and connect to Teleport's web port directly to isolate the issue.

Reviewer: Does "connect" refer to client connections or connecting Teleport services? Does "Teleport" refer to the Proxy Service?

Author: Both. I'll clarify.

To use Teleport behind a reverse proxy, you should either:
- use a layer 4 (TCP) proxy which forwards TCP streams directly to Teleport (which will in turn handle TLS termination itself)
- disable Teleport's TLS routing mode by adding `version: v1` to your config file and removing `proxy_listener_mode: multiplex`

>   - You can get an example v1 config file using `teleport configure --version=v1 --public-addr=teleport.example.com:443` (change the public address to your own domain)

Reviewer: ```suggestion
 
You can get an example v1 config file using `teleport configure --version=v1 --public-addr=teleport.example.com:443` (change the public address to your own domain)
```
Since this is a new thought rather than a continuation of the list (it doesn't follow "you should either:")

>   - A list of ports which will be used can be found here: [ports without TLS routing](../reference/networking.mdx#ports-without-tls-routing)

Reviewer: ```suggestion
If disabling TLS routing, you can find a list of ports to use for connecting different Teleport services at [ports without TLS routing](../reference/networking.mdx#ports-without-tls-routing)
```
To make this a bit more specific
